### DISCUSSION TITLE
CDDA testing

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#43] [Show and tell]
For compaq 6910p, cdrom drive and media are detected, but I can't see the audio cd in the media plaver.

[compaq6910p_audioCD_fail.txt](https://github.com/VendelinSlezak/BleskOS/files/13854817/compaq6910p_audioCD_fail.txt)

**Comment by defdefred** [comment]
For Packard-Bell, I can see the audio CD, but reading track is failing.
I tried read-skipping-bad sector and all sector were bad (you should add a cancel possibility and maybe a mouse move between each sector).
Waiting now for the end of the track reading to see system log is giving interesting data...

**Reply by defdefred** [reply]
I can't show you the cdda file coz PrtScr key is not making a screenshoot with my hp usb keyboard.
[packardbellcddareadfail.txt](https://github.com/VendelinSlezak/BleskOS/files/13854989/packardbellcddareadfail.txt)

**Comment by VendelinSlezak** [comment]
Thank you for testing this feature, these things should be fixed in update 6. Also when you insert CD into drive, it take some time for drive to recognize it, so after insertion you should wait until drive is not reading disk. BleskOS will not recognize inserted CD automatically so you then need to press [F8] Refresh devices even several times until CD show up.

**Comment by defdefred** [comment]
Will do.
I have usb2ide adaptor where I can test cdrom drive.
Is it useful or not yet implemented?

**Comment by defdefred** [comment]
Same with u6
[packardbellcddareadfail_2.txt](https://github.com/VendelinSlezak/BleskOS/files/13866587/packardbellcddareadfail_2.txt)
Same with another cdrom drive
[packardbell_anothercdromdrive.txt](https://github.com/VendelinSlezak/BleskOS/files/13866590/packardbell_anothercdromdrive.txt)

And it is weird but bleskOS is detecting a touchpad but it is a desktop pc...
![packardbell_nonexisttouchpad](https://github.com/VendelinSlezak/BleskOS/assets/17281367/31545f21-73ba-48c4-89f9-e1aef9631d84)

**Comment by VendelinSlezak** [comment]
I tested update 7 on 5 real computers and it worked everywhere so I hope it will work also on your computers. And BleskOS shows touchpad, because it detects PS/2 mouse. Because touchpad is connected to PS/2 mouse port and 99,9% users do not use PS/2 mouse, BleskOS assume that every PS/2 mouse is touchpad.

**Comment by defdefred** [comment]
cdda reading and saving ok
[packardbellcddaok.txt](https://github.com/VendelinSlezak/BleskOS/files/13881097/packardbellcddaok.txt)
Well-done!

**Comment by defdefred** [comment]
I noticed that the cdrom drive is very noisy while playing the cdda. Is it possible to set the speed to x1?

**Comment by VendelinSlezak** [comment]
When you play CDDA, BleskOS is not reading anything from CD, it plays file from memory. However when you request some sector from CD, it will automatically load more sectors to internal memory to provide faster further reading. When you read larger file, CD will be spinned up to high rotations so even when you stop reading, it will work for while to read some more sectors. I fixed this in Update 8, where after reading whole file it spins disk down, so after return to program drive is not rotating disk so there is no noise.
